### DISCUSSION TITLE
'Get' method brief documentation.

### DISCUSSION THREAD

**Discussion by Aman-singhmar** [react-hook-form/react-hook-form#12032] [Q&A]
Hi,
I am using the react hook form 'get' method import like:
import {  get } from 'react-hook-form';

But I do not find anything about 'get' in the official document. Can you provide me a document/ link for this?

**Comment by asheeshSharma2000** [comment]
Hey, i'm also confused related to the working of this method, please provide some documentation or any source where i can read about it as i didn't found anything in the documentation.

**Comment by tatsmaki** [comment]
It is used internally, so there are no docs. You can expect that it can be changed in future versions.
Few points to keep in mind:
1) it is better to use it with TypeScript
2) it can get value of an object using complex path
3) there is also `set` function used to set complex field value

![Screenshot 2024-06-19 at 13 59 35](https://github.com/react-hook-form/react-hook-form/assets/61022192/ce24bc7a-29ed-432c-ac1c-5637da9dc341)

**Reply by Aman-singhmar** [reply]
Thanks !

**Comment by asheeshSharma2000** [comment]
So you mean to say that we shouldn't use this method ?

**Reply by tatsmaki** [reply]
I think we can, it is quite useful for nested fields
You can also try to open PR to improve documentation https://github.com/react-hook-form/documentation

**Comment by asheeshSharma2000** [comment]
Okay , thanks for this information.
